### PR TITLE
Bluetooth lyric

### PR DESCRIPTION
**Pull request by tramhao**
Send lyric to car connected with bluetooth.
Is optional and depend on the car model for displaying the message. 
Tested on toyota hilander old version and looks good.
However, about split one line of lyric to different lines(added as StringUtil.wideCharacterSplit), in English, is there some good way to avoid split a word? Suggestions needed.

### CODE REVIEW

File: app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java [hunk 1]
```diff
@@ -619,6 +620,35 @@ public void onResourceReady(@NonNull Bitmap resource, Transition<? super Bitmap>
         }
     }
 
+    public void updateMediaSessionLyric(String line) {
+        final Song song = getCurrentSong();
+
+        if (song.id == -1) {
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java:626]: Better use `song.id == s=Song.EMPTY.id`, ie dont assume implicitly the use of -1 constant.

File: app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java [hunk 2]
```diff
@@ -619,6 +620,35 @@ public void onResourceReady(@NonNull Bitmap resource, Transition<? super Bitmap>
...
+            mediaSession.setMetadata(null);
+            return;
+        }
+
+        List<String> splitlist = StringUtil.wideCharacterSplit(line,20);
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java:631]: what is 20 please?

File: app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java [hunk 3]
```diff
@@ -619,6 +620,35 @@ public void onResourceReady(@NonNull Bitmap resource, Transition<? super Bitmap>
...
+
+        // set Artist, Title and Album to lyrics
+        final MediaMetadataCompat.Builder metaData = new MediaMetadataCompat.Builder()
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java:634]: I dont get it: we are not sending lyrics here.

And if I'm not wrong, there is already similar code that publishes the now-playing metadata for the other apps (Android Auto, musixmatch, lock screen, ...)

Comment (soncaokim) [reply app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java:634]: Are you trying to use the artist/title/album to convey the first 3 lines of the synchronized lyrics?

**Review by soncaokim — COMMENTED** [review]
I oppose to merge this change, for two reasons:
- If my understanding is correct, you are sending lyrics (3 lines of it) via fields that are there for another purpose.
- I dont see it very useful and even dangerous (hence not motivated to maintain) to show lyrics in a driving environment.

File: app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java (hunk 2 above)
Comment (tramhao) [reply app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java:631]: In car display, the space is limited to 20 letters, so I need to split the line as in many cases, the line is longer than 20 letters.

File: app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java (hunk 3 above)
Comment (tramhao) [reply app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java:634]: Yes, I get the idea from several apps(like netease-music and qq-music) but they only use artist as lyric, so they cannot display full lyrics.

**Comment by tramhao** [comment]
> * s correct, you are sending lyrics (3 lines of it) via fields that are there for another purpose.
> * I dont see it very useful and even dangerous (hence not motivated to maintain) to show lyrics in a

This could be dangerous, but normally I'll stair at gps anyway:)
Don't worry, if you think it's not appropriate, just close it and I'll keep it in my fork.

**Comment by tramhao** [comment]
Personally I use this function because when I drive with my daughter, she likes to see the lyric but I need phone for gps. With this, she can see the lyrics in center display.

**Comment by tramhao** [comment]
For those interested in this function, please check out in my repo under bluetooth-lyric branch. I'll close this for now and keep maintaining that branch. Thanks.

**Pull request closed without merging**